src/libcore/sync/atomic.rs

```diff
@@ -310,11 +310,103 @@ impl AtomicBool {
     /// ```
     #[inline]
     #[stable(feature = "rust1", since = "1.0.0")]
+    #[rustc_deprecated(reason = "replaced with compare_exchange", since = "1.8.0")]
     pub fn compare_and_swap(&self, current: bool, new: bool, order: Ordering) -> bool {
+        self.compare_exchange(current, new, order, strongest_failure_ordering(order))
+    }
+
+    /// Stores a value into the `bool` if the current value is the same as the `current` value.
+    ///
+    /// The return value is always the previous value. If it is equal to `current`, then the value
+    /// was updated.
+    ///
+    /// `compare_exchange` takes two `Ordering` arguments to describe the memory ordering
+    /// of this operation. The first describes the required ordering if the operation succeeds while
+    /// the second describes the required ordering when the operation fails. The failure ordering
+    /// can't be `Acquire` or `AcqRel` and must be equivalent or weaker than the success ordering.
```

Comment:
Actually the surrounding style seems to wrap comments at 100 chars, which is what I've been using.